source/MarlinProcessors/src/DrawBeamCalFromDD4hep.cpp

```diff
@@ -331,9 +331,8 @@ void DrawBeamCalFromDD4hep::drawCartesianGridXY() {
   for (MapIdVal::iterator it = hitEnergies.begin(); it != hitEnergies.end(); ++it) {
     unsigned int cellid = it->first;
     double energy = it->second;
-    (*m_seg.segmentation()->decoder()).setValue(0, cellid);
-    int xBin = (*m_seg.segmentation()->decoder())["x"];
-    int yBin = (*m_seg.segmentation()->decoder())["y"];
+    int xBin = m_seg.segmentation()->decoder()->get(cellid, "x");
```

Comment:
I don't think this is correct, you have to convert `cellid` to `long long`, don't you? because cellid is cellid1 from the calorimeterhit.

Reply:
yep, will fix it.

Reply:
In was already working on this as well.
But this might be a major head-ache in the future when people pass the wrong type to this function.